gson/src/test/java/com/google/gson/functional/NamingPolicyTest.java

```diff
@@ -237,6 +238,78 @@ public void testAtSignInSerializedName() {
     assertThat(new Gson().toJson(new AtName())).isEqualTo("{\"@foo\":\"bar\"}");
   }
 
+  @Test
+  public void testGsonWithNameDeserialiation() {
+    Gson gson =
+        builder
+            .setFieldNamingStrategy(
+                new FieldNamingStrategy() {
+
+                  @Override
+                  public String translateName(Field f) {
+                    return "SomeConstantStringInstanceField";
+                  }
+
+                  @Override
+                  public List<String> alternateNames(Field f) {
+                    return List.of("alternate-name");
+                  }
+                })
+            .create();
+    String target = "{\"SomeConstantStringInstanceField\":\"someValue\"}";
```

Comment:
Similar to the other comment, use a custom name which differs more from the real field name.
```suggestion
                    return "primary-name";
                  }

                  @Override
                  public List<String> alternateNames(Field f) {
                    return List.of("alternate-name");
                  }
                })
            .create();
    String target = "{\"primary-name\":\"someValue\"}";
```